packages/bun/src/integrations/bunsqlite.ts

```
@@ -0,0 +1,216 @@
+          return;
+        }
+
+        proto[method] = function (this: any, ...args: any[]) {
+          let dbName = this._sentryDbName || dbNameMap.get(this);
+
+          if (!dbName && this.filename) {
+            dbName = this.filename;
+            dbNameMap.set(this, dbName);
+          }
+
+          const sql = method !== 'transaction' && args[0] && typeof args[0] === 'string' ? args[0] : undefined;
+
+          if (inParentSpanMap.get(this) && method === 'prepare') {
+            const result = originalMethod.apply(this, args);
+            if (result) {
+              return instrumentStatement(result, sql, dbName);
+            }
+            return result;
+          }
+
+          return startSpan(
+            {
+              name: sql || 'db.sql.' + method,
+              op: `db.sql.${method}`,
+              attributes: {
+                [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.db.bun.sqlite',
+                'db.system': 'sqlite',
+                'db.operation': method,
+                ...(sql && { 'db.statement': sql }),
+                ...(dbName && { 'db.name': dbName }),
+              },
+            },
+            span => {
+              try {
+                const wasInParentSpan = inParentSpanMap.get(this) || false;
+                if (method === 'query') {
+                  inParentSpanMap.set(this, true);
+                }
+
+                const result = originalMethod.apply(this, args);
+
+                if (wasInParentSpan) {
+                  inParentSpanMap.set(this, wasInParentSpan);
+                } else {
+                  inParentSpanMap.delete(this);
+                }
+
+                if (method === 'prepare' && result) {
+                  return instrumentStatement(result, sql, dbName);
+                }
+
+                return result;
+              } catch (error) {
+                span.setStatus({ code: 2, message: 'internal_error' });
+                captureException(error, {
```

We don't need to capture an error here, we can just let the error bubble up.